### DISCUSSION TITLE
Swagger / Redoc documentation

### DISCUSSION THREAD

**Discussion by SylvainCorlay** [uNetworking/uWebSockets#1142] [Ideas]
Hello, uwebsockets looks like a perfect fit for a project where I need to process lots of websockets messages efficiently. Thanks for the amazing work!

On the rest endpoints side, is there any example out there of people generating and hosting swagger or redoc documentation for a uwebsockets-based API?

**Comment by ghost** [comment]
I know even less about that than you do. I have no idea - probably not?

**Comment by SylvainCorlay** [comment]
Thank you for your reply!

I don't know how we could achieve something like this, but it would be a really compelling selling point for uwebsockets. (I am personally convinced already).

**Comment by ghost** [comment]
I don't even know exactly what swagger is? You generate code with it? And documentation?

That probably works for hugely popular things with very little requirements variation - simple repetitive things?

But it still kind of sounds like doing the same work only in different ways?

**Comment by SylvainCorlay** [comment]
Sorry for the late reply. Swagger is a tool to document REST APIS, and provides a UI to interact with a running server. You can check out e.g. https://petstore.swagger.io/.

For some frameworks like FastAPI, the swagger JSON file is automatically generated from the signatures of the endpoints.

**Comment by ghost** [comment]
What I can see there would have to be alone kind of interoperability with "OpenAPI Specification"?

That could probably be implemented as an abstraction around the uWS::App without touching anything internal

**Comment by ghost** [comment]
Like SwaggerApp inheriting from uWS::App

**Comment by SylvainCorlay** [comment]
The real deal would be to generate them somewhat automatically.

**Comment by ghost** [comment]
Well yes but you have to implement this somewhere and it doesn't belong with anything internal. It's an abstraction eating/consuming this specification.

**Comment by brunoqs** [comment]
Hello im using this lib to generate openapi.json https://www.npmjs.com/package/swagger-jsdoc?activeTab=readme. But the i want to create swagger routes also
